packages/tasks/test/swap/UniswapV2Swapper.test.ts

```diff
@@ -0,0 +1,464 @@
+                  const message = ethers.utils.solidityKeccak256(['bytes'], [encodedPrices])
+                  const signature = await owner.signMessage(ethers.utils.arrayify(message))
+                  const data = defaultAbiCoder.encode([PricesDataType, 'bytes'], [pricesData, signature]).slice(2)
+                  const dataLength = defaultAbiCoder.encode(['uint256'], [data.length / 2]).slice(2)
+                  extraCallData = `${data}${dataLength}`
+                })
+
+                beforeEach('set threshold', async () => {
+                  const setDefaultTokenThresholdRole = task.interface.getSighash('setDefaultTokenThreshold')
+                  await authorizer
+                    .connect(owner)
+                    .authorize(owner.address, task.address, setDefaultTokenThresholdRole, [])
+                  await task.connect(owner).setDefaultTokenThreshold(tokenOut.address, thresholdAmount, 0)
+                })
+
+                const executeTask = async (amountIn, slippage, hopTokens): Promise<ContractTransaction> => {
+                  const callTx = await task.populateTransaction.call(tokenIn.address, amountIn, slippage, hopTokens)
+                  const callData = `${callTx.data}${extraCallData}`
+                  return owner.sendTransaction({ to: task.address, data: callData })
+                }
+
+                context('when the smart vault balance passes the threshold', () => {
+                  let hopToken1: Contract
+                  let hopToken2: Contract
+
+                  beforeEach('deploy hop tokens', async () => {
+                    hopToken1 = await deployTokenMock('HOP_TKN_1')
+                    hopToken2 = await deployTokenMock('HOP_TKN_2')
+                  })
+
+                  beforeEach('fund smart vault', async () => {
+                    await tokenIn.mint(smartVault.address, amountIn)
+                  })
+
+                  context('when the slippage is below the limit', () => {
+                    const hopTokens = []
+                    const slippage = fp(0.01)
+                    const expectedAmountOut = amountIn.mul(tokenRate)
+                    const minAmountOut = expectedAmountOut.mul(fp(1).sub(slippage)).div(fp(1))
+
+                    beforeEach('set max slippage', async () => {
+                      const setDefaultMaxSlippageRole = task.interface.getSighash('setDefaultMaxSlippage')
+                      await authorizer
+                        .connect(owner)
+                        .authorize(owner.address, task.address, setDefaultMaxSlippageRole, [])
+                      await task.connect(owner).setDefaultMaxSlippage(slippage)
+                    })
+
+                    const itExecutesTheTaskProperly = (requestedAmount: BigNumberish) => {
+                      it('executes the expected connector', async () => {
+                        const hopTokens = [hopToken1.address, hopToken2.address]
```

Comment:
Since the hopTokens are being used here I'd do this:

```suggestion
                      let hopTokens: string[]
                      
                      beforeEach('deploy hop tokens', async () => {
                        const hopToken1 = await deployTokenMock('HOP_TKN_1')
                        const hopToken2 = await deployTokenMock('HOP_TKN_2')
                        hopTokens = [hopToken1.address, hopToken2.address]
                      })
                      
                      it('executes the expected connector', async () => {
```